src/routing/dispatcher.rs

```diff
@@ -42,6 +42,10 @@ impl Dispatcher {
         }
     }
 
+    pub async fn contains_incoming(&self, msg_id: &MessageId) -> bool {
+        self.core.lock().await.contains_incoming(msg_id)
+    }
+
```

Comment:
Possibly better to have `add_to_filter() ->bool` if added then cool, if not added then drop message (it's filtered)